### DISCUSSION TITLE
Visualizing bbox predictions/proposals on images (Tensorboard) during training?

### DISCUSSION THREAD

**Discussion by cpereir1** [facebookresearch/detectron2#3577] [Q&A]
Hi!

Has anyone implemented visualizing the predictions (bbox in this case) overlaid on images, for display for example in Tensorboard, while training/fine-tuning a detectron2 model using train_net.py (for example)? This would be very useful to have, and see it is not there out-of-the-box. Hoping someone has had the same need and could share his/hers implementation.

Many thanks and great work so far!

**Comment by sukjunhwang** [comment]
I believe the function below can help you.

https://github.com/facebookresearch/detectron2/blob/31ec19b3132a3ac609600802dd37b2b40a76b5c9/detectron2/modeling/meta_arch/rcnn.py#L87

**Reply by Shuntw6096** [reply]
You can set ```VIS_PERIOD``` in config file, it will do visualization every few iterations during training.
https://github.com/facebookresearch/detectron2/blob/0e6702d8c5589f434efb054400869826b5218956/detectron2/config/defaults.py#L624
